modules/dnn/src/layers/const_layer.cpp

```diff
@@ -62,10 +62,15 @@ class ConstLayerImpl CV_FINAL : public ConstLayer
     {
         std::vector<UMat> outputs;
         outs.getUMatVector(outputs);
-        if (outs.depth() == CV_16S)
-            convertFp16(blobs[0], outputs[0]);
+        if (outs.depth() == CV_16S) {
+            auto blob = blobs[0];
+            if (blob.type() == CV_32S) {
```

Comment:
`if (blob.type() != CV_32F)`

Reply:
I found that `convertFp16` can do two conversion, fp32 -> fp16 and fp16 -> fp32. Won't it break if `blob.type() == CV_16F`?

Reply:
Agree, but seems like it didn't work with previous code too. Not sure if `CV_16F` is even supported in dnn.

Reply:
Yes, we do support `CV_16F` for CPU target. We have conv kernels for fp16 and also have real fp16 models in model zoo, https://github.com/opencv/opencv_zoo/tree/main/models/text_recognition_crnn. Also we are trying to convert all models in model zoo to fp16 accuracy, https://github.com/opencv/opencv_zoo/pull/183.

Reply:
This is `forward_ocl` method, not CPU.